src/server/notifications/detail-fetchers/buzz.detail-fetcher.ts

```diff
@@ -0,0 +1,26 @@
+import { createDetailFetcher } from '~/server/notifications/detail-fetchers/base.detail-fetcher';
+import { buzzNotifications } from '~/server/notifications/buzz.notifications';
+import { simpleUserSelect } from '~/server/selectors/user.selector';
+import { isDefined } from '~/utils/type-guards';
+
+export const buzzDetailFetcher = createDetailFetcher({
+  types: [...Object.keys(buzzNotifications)],
+  fetcher: async (notifications, { db }) => {
+    const userIds = notifications
+      .map((n) => n.details.fromUserId as number | undefined)
+      .filter(isDefined);
+    if (userIds.length === 0) return;
+
+    const users = await db.user.findMany({
+      where: { id: { in: userIds } },
+      select: simpleUserSelect,
+    });
+
+    for (const n of notifications) {
+      const user = users.find((c) => c.id === n.details.fromUserId);
+      if (user) {
+        n.details.actor = user;
+      }
+    }
+  },
+});
```

Comment:
iirc, buzz notifications redirect the user to the entity that was given buzz (i.e, model, image, etc). Might be misremembering there but worth to point out

Reply:
Yes that's still how it works, but this is notifications for tipping. This is just fetching extra data to display the avatar of the user who tipped you